### PR TITLE
feat: Basic `ListView` keyboard navigation

### LINKED ISSUE #7732: [WASM] ListViewItem focus issues
### Current behavior

Add a ListView to a view.  Enable Local tab navigation (TabNavigation="Local") so each item can receive tab focus.  The following issues were found in WASM which work in UWP:
1. The items do not have their focus visual states applied.
2. Space bar does not select the item.

UWP:
image

WASM:
image

### Expected behavior

The focus visual states are applied and the user can select with spacebar.

### How to reproduce it (as minimally and precisely as possible)

Attaching a sample sln.
[App1.Shared.zip](https://github.com/unoplatform/uno/files/7803599/App1.Shared.zip)

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

    <PackageReference Include="Uno.Material" Version="1.0.3" />
    <PackageReference Include="Uno.Microsoft.Toolkit.Uwp.UI.Controls" Version="7.1.0-build.45.gd2e4192c8c" />
    <PackageReference Include="Uno.Microsoft.Toolkit.Uwp.UI.Controls.DataGrid" Version="7.1.0-build.45.gd2e4192c8c" />
    <PackageReference Include="Uno.Microsoft.Xaml.Behaviors.Uwp.Managed" Version="2.0.3-uno.95" />
    <PackageReference Include="Uno.UI.WebAssembly" Version="3.11.6" />
    <PackageReference Include="Uno.UI.RemoteControl" Version="3.11.6" Condition="'$(Configuration)'=='Debug'" />
    <PackageReference Include="Uno.Wasm.Bootstrap" Version="3.0.0" />
    <PackageReference Include="Uno.Wasm.Bootstrap.DevServer" Version="3.0.0" />

### Affected platforms

WebAssembly

### IDE

Visual Studio 2022

### IDE version

Version 17.1.0 Preview 1.0

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #7732

## PR Type

What kind of change does this PR introduce?

- Bugfix
- Feature

## What is the current behavior?

- Can't navigate/select `ListView` items
- Focus styles not applied

## What is the new behavior?

- Can navigate/select `ListView` items
- **TODO** Focus styles not applied

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ListView/ListView_Focus.xaml [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                                         ContentTemplate="{TemplateBinding ContentTemplate}"
+                                         ContentTemplateSelector="{TemplateBinding ContentTemplateSelector}"
+                                         ContentTransitions="{TemplateBinding ContentTransitions}"
+                                         Feedback="{ThemeResource PrimaryHueMidBrush}"
+                                         FeedbackOpacity="{StaticResource MaterialPressedOpacity}"
+                                         FontFamily="{TemplateBinding FontFamily}"
+                                         FontSize="{TemplateBinding FontSize}"
+                                         FontWeight="{TemplateBinding FontWeight}"
+                                         Foreground="{TemplateBinding Foreground}"
+                                         IsTabStop="False" />-->
```
Comment (jeromelaban) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ListView/ListView_Focus.xaml:55]:
```suggestion
```

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ListView/ListView_Focus.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+		public ListView_Focus()
+		{
+			this.InitializeComponent();
+
+			DataContext = ViewModel = new ListViewFocusViewModel();
+			for (int i = 0; i < 3; i++)
+			{
+				var item = new ListViewFocusViewModelItem { Label = $"Test{i}" };
+				//item.Items.Add(new TestViewModelItem { Label = "Loading..." });
```
Comment (jeromelaban) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ListView/ListView_Focus.xaml.cs:23]:
```suggestion
```

File: src/Uno.UI/UI/Xaml/Controls/Primitives/VisualStatesHelper.mux.cs [hunk 3]
```diff
@@ -10,31 +10,31 @@ internal static IEnumerable<string> GetValidVisualStatesListViewBaseItem(ListVie
 		{
 			int index = 0;
 			//int expectedVisualStatesSize = 6;
-			int expectedVisualStatesSize = 1; // See Uno TODO below
+			int expectedVisualStatesSize = 2; // See Uno TODO below
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/Primitives/VisualStatesHelper.mux.cs:13]: Is there still a todo?

**Review by jeromelaban — COMMENTED** [review]
Are you able to create UI tests for this?

File: src/Uno.UI/UI/Xaml/Controls/Primitives/VisualStatesHelper.mux.cs (hunk 3 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/Primitives/VisualStatesHelper.mux.cs:13]: Yes, the remaining states need to be added later as well, but probably when we fully port list items and selector items code. The final size should be 6 👍

**Review by jeromelaban — APPROVED** [review]
Very nice !!

**Review by kazo0 — APPROVED** [review]

**Review by carldebilly — APPROVED** [review]

**Pull request merged**